patches/server/0067-lithium-skip-ticking-block-entities-that-are-doing-n.patch

```diff
@@ -0,0 +1,1348 @@
+index b0fdd5e3eae999b515a2eb62eb24d6d66c61f5f5..17b974c6a0aeaf5114889bb7cfb82880f2089901 100644
+--- a/src/main/java/net/minecraft/server/TileEntityShulkerBox.java
++++ b/src/main/java/net/minecraft/server/TileEntityShulkerBox.java
+@@ -8,6 +8,7 @@ import javax.annotation.Nullable;
+ import org.bukkit.craftbukkit.entity.CraftHumanEntity;
+ import org.bukkit.entity.HumanEntity;
+ // CraftBukkit end
++import me.jellysquid.mods.lithium.common.world.blockentity.BlockEntitySleepTracker;
+ 
+ public class TileEntityShulkerBox extends TileEntityLootable implements IWorldInventory, ITickable {
+ 
+@@ -16,7 +17,9 @@ public class TileEntityShulkerBox extends TileEntityLootable implements IWorldIn
+     public int viewingCount;
+     private TileEntityShulkerBox.AnimationPhase i;
+     private float j;
++    public float animationProgress = j; // Yatopia - OBFHELPER
+     private float k;
++    public float prevAnimationProgress = k; // Yatopia - OBFHELPER
+     @Nullable
+     private EnumColor l;
+     private boolean m;
+@@ -71,6 +74,11 @@ public class TileEntityShulkerBox extends TileEntityLootable implements IWorldIn
+             this.m();
+         }
+ 
++        // Yatopia start - lithium: skip ticking block entities that are doing nothing
++        if (this.getAnimationPhase() == TileEntityShulkerBox.AnimationPhase.CLOSED && this.prevAnimationProgress == 0f && this.animationProgress == 0f && this.world != null) {
++            ((BlockEntitySleepTracker) this.world).setAwake(this, false);
++        }
++        // Yatopia end
+     }
+ 
+     protected void h() {
+@@ -102,6 +110,7 @@ public class TileEntityShulkerBox extends TileEntityLootable implements IWorldIn
+ 
+     }
+ 
++    public TileEntityShulkerBox.AnimationPhase getAnimationPhase() {return this.j(); } // Yatopia - OBFHELPER
+     public TileEntityShulkerBox.AnimationPhase j() {
+         return this.i;
+     }
+@@ -184,6 +193,7 @@ public class TileEntityShulkerBox extends TileEntityLootable implements IWorldIn
+ 
+     @Override
+     public boolean setProperty(int i, int j) {
++        if (this.world != null && i == 1) ((BlockEntitySleepTracker) this.world).setAwake(this, true); // Yatopia
+         if (i == 1) {
+             this.viewingCount = j;
+             if (j == 0) {
+diff --git a/src/main/java/net/minecraft/server/TileEntitySkull.java b/src/main/java/net/minecraft/server/TileEntitySkull.java
+index ab553f408b31cb7e2d281cc3d8fe42e19dc717d9..6e426d8e4c6e3768730b0d66267f44c1eee645ed 100644
+--- a/src/main/java/net/minecraft/server/TileEntitySkull.java
++++ b/src/main/java/net/minecraft/server/TileEntitySkull.java
+@@ -23,6 +23,7 @@ import com.mojang.authlib.Agent;
+ import com.mojang.authlib.ProfileLookupCallback;
+ import java.util.concurrent.Callable;
+ // Spigot end
++import me.jellysquid.mods.lithium.common.world.blockentity.BlockEntitySleepTracker;
+ 
+ public class TileEntitySkull extends TileEntity /*implements ITickable*/ { // Paper - remove tickable
```

Comment:
Is it necessary to do extra logic if skull isnt tickable?

Reply:
Nope